diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 1fd2a2eb93a80150393aaf1b6a1e3fc295bf7784..0000000000000000000000000000000000000000
diff --git a/lib/Icon.astro b/lib/Icon.astro
index cfd0d9cb0dd5ee71b3e34f885e7df70b418a99be..65c5b85382bea3439818cc86a46204af4f42d1f7 100644
--- a/lib/Icon.astro
+++ b/lib/Icon.astro
@@ -1,6 +1,6 @@
 ---
 import load, { fallback, normalizeProps } from './utils.ts';
-export { Props } from './Props.ts';
+export type { Props } from './Props.ts';
 
 let { name, pack, title, optimize = true, class: className, ...inputProps } = Astro.props;
 let props = {};
